contracts/core/Allo.sol

```diff
@@ -513,24 +427,72 @@ contract Allo is Transfer, Initializable, Ownable, AccessControl {
     }
 
     /// ====================================
-    /// ======= Internal Functions =========
+    /// ======= Strategy Functions =========
     /// ====================================
 
+    /// @notice passes _data through to the allocation strategy for that pool
+    /// @notice returns the recipientId from the allocation strategy
+    /// @param _poolId id of the pool
+    /// @param _data encoded data unique to the allocation strategy for that pool
+    function registerRecipients(uint256 _poolId, bytes memory _data) external payable returns (address) {
+        return pools[_poolId].strategy.registerRecipients(_data, msg.sender);
+    }
+
+    /// @notice Fund a pool
+    /// @param _poolId id of the pool
+    /// @param _amount extra amount of the token to be deposited into the pool
+    /// @param _token The address of the token that the pool is denominated in
+    /// @dev Anyone can fund a pool
+    function fundPool(uint256 _poolId, uint256 _amount, address _token) external payable {
+        if (_amount == 0) {
+            revert NOT_ENOUGH_FUNDS();
+        }
+        _fundPool(_token, _amount, _poolId, pools[_poolId].strategy);
+    }
+
     /// @notice passes _data & msg.sender through to the allocation strategy for that pool
     /// @param _poolId id of the pool
     /// @param _data encoded data unique to the allocation strategy for that pool
+    function allocate(uint256 _poolId, bytes memory _data) external payable {
+        _allocate(_poolId, _data);
+    }
+
+    /// @notice vote to multiple pools
+    /// @param _poolIds ids of the pools
+    /// @param _datas encoded data unique to the allocation strategy for that pool
+    function batchAllocate(uint256[] calldata _poolIds, bytes[] memory _datas) external {
+        for (uint256 i = 0; i < _poolIds.length;) {
+            _allocate(_poolIds[i], _datas[i]);
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice passes _data & msg.sender through to the disribution strategy for that pool
+    /// @param _poolId id of the pool
+    /// @param _data encoded data unique to the strategy for that pool
+    function distribute(uint256 _poolId, address[] memory _recipientIds, bytes memory _data) external {
+        pools[_poolId].strategy.distribute(_recipientIds, _data, msg.sender);
+    }
+
+    /// ====================================
+    /// ======= Internal Functions =========
+    /// ====================================
+
+    /// @notice passes _data & msg.sender through to the strategy for that pool
+    /// @param _poolId id of the pool
+    /// @param _data encoded data unique to the strategy for that pool
     function _allocate(uint256 _poolId, bytes memory _data) internal {
-        pools[_poolId].allocationStrategy.allocate{value: msg.value}(_data, msg.sender);
+        pools[_poolId].strategy.allocate{value: msg.value}(_data, msg.sender);
     }
```

Comment:
should we check if the pool exists?

Reply:
wouldn't it just fail is pool doesn't exist ?

Reply:
If it doesn't exist, strategy will be `address(0)` so it'll revert. We can check if you want custom error message in that case.

Reply:
i think it's fine @KurtMerbeth 
we could add that in later if we feel strongly about it